### PR TITLE
`azurerm_storage_account`: Add support for `AzureDNSZone`

### PR DESCRIPTION
**Pull request by favoretti**
```
TF_ACC=1 go test -v ./internal/services/storage -timeout=1000m -run='TestAccStorageAccount_DNSEndpointTypeAzure'
=== RUN   TestAccStorageAccount_DNSEndpointTypeAzure
=== PAUSE TestAccStorageAccount_DNSEndpointTypeAzure
=== CONT  TestAccStorageAccount_DNSEndpointTypeAzure
--- PASS: TestAccStorageAccount_DNSEndpointTypeAzure (150.99s)
PASS
ok  	github.com/hashicorp/terraform-provider-azurerm/internal/services/storage	152.828s
```

### CODE REVIEW

**Comment by stephybun** [comment]
@favoretti is this ready for review? Or could we close until it's ready for review?

**Comment by favoretti** [comment]
@stephybun im trying to catch @magodo on slack, since he was apparently working on the same issue - I wanted to collaborate and get this done. We need this badly, I'm just a bit lost in Giovanni and underlying changes that are needed for this to be finalized.

**Comment by katbyte** [comment]
@favoretti - whats the status on this? are yous till trying to reach out to @magodo or?

**Comment by magodo** [comment]
I believe this depends on https://github.com/tombuildsstuff/giovanni/pull/78

**Comment by favoretti** [comment]
@magodo @katbyte ill rerun the tests tomorrow and will check if it all works. If it does - I'll convert to a normal PR. Sorry I disappeared lately - too much stuff going on work/life wise.

**Comment by tombuildsstuff** [comment]
Dependent on https://github.com/tombuildsstuff/giovanni/issues/68

**Comment by tombuildsstuff** [comment]
Dependent on https://github.com/hashicorp/terraform-provider-azurerm/pull/24213

**Comment by favoretti** [comment]
Moved dependent on to #24798

**Comment by favoretti** [comment]
@tombuildsstuff @katbyte @stephybun yay. local acceptance passes, kicked TC, but otherwise ready for a review!

**Comment by favoretti** [comment]
@tombuildsstuff apologies, didn't notice I pushed an argument-specific note. Fixed.

File: website/docs/r/storage_account.html.markdown [hunk 1]
```diff
@@ -173,6 +173,8 @@ The following arguments are supported:
 
 -> **NOTE:** SFTP support requires `is_hns_enabled` set to `true`. [More information on SFTP support can be found here](https://learn.microsoft.com/azure/storage/blobs/secure-file-transfer-protocol-support). Defaults to `false`
 
+* `dns_endpoint_type` - (Optional) Which DNS endpoint type to use. Possible values are `Standard` and `AzureDnsZone`. Defailts to `Standard`. Changing this forces a new resource to be created.
```
Comment (tombuildsstuff) [inline website/docs/r/storage_account.html.markdown:176]: Just spotted this now I'm looking closer, apologies:

```suggestion
* `dns_endpoint_type` - (Optional) Specifies which DNS endpoint type to use. Possible values are `Standard` and `AzureDnsZone`. Defaults to `Standard`. Changing this forces a new resource to be created.
```

**Review by tombuildsstuff — APPROVED** [review]
One nit but otherwise 👍

**Pull request merged**